rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,693 @@
+  Kibana instance (ie: results from another space, results from another Kibana instance in a cloud cluster)
+
+However, as forging absolute urls can be a tedious process for the plugins, the `url` property of results returned by
+a result provider can be either an absolute url or an url path relative to the executing request's `basePath`.
+
+I.E are considered valid:
+- `https://my-other-kibana-instance/some/result`
+- `/app/kibana#/dashboard/some-id`
+
+When consolidating the result, the logic regarding the `url` property is:
+- if the `url` is absolute, return it unchanged
+- if the `url` is a relative path (starts with `/`), it will be converted to an absolute url by prepending the Kibana
+instance's newly introduced `publicAddress`.
+
+#### server.publicAddress
+
+Given the fact that some Kibana deployments have complex architecture (proxies, rewrite rules...), there is currently
+no reliable way to know for sure what the public address used to access kibana is (at least from the server-side).
+
+A new `server.publicAddress` property will be added to the kibana configuration, allowing ops to explicitly define the public
+address to instance is accessible from. This property is only meant to be used to generate 'links' to arbitrary resources
+and will not be used to configure the http server in any way.
+
+When not explicitly defined, this property will be computed using the known `server` configuration values:
+
+```ts
+const defaultPublicAddress = removeTrailingSlash(
+  `${getServerInfo().protocol}://${httpConfig.host}:${httpConfig.port}/${httpConfig.basePath}`
+);
+
+const getPublicAddress = () => httpConfig.publicAddress ?? defaultPublicAddress;
+```
+
+A new `getAbsoluteUrl` api will also be added to the core `http` service contract:
+
+```ts
+const getAbsoluteUrl = (path: string, request: KibanaRequest) => {
+  const publicUrl = getPublicAddress();
+  const absoluteUrl = joinRemovingDuplicateAndTrailingSlash(
+    publicUrl,
+    // note: this is actually wrong. We would need the `requestScopePath` here
+    // as this currently returns `${this.serverBasePath}${requestScopePath}` and the basePath
+    // is already included in `publicAddress`
+    serverContract.basePath.get(request), 
+    path
+  );
+}
+```
+
+Search results will then be consolidated before being returned to convert relative urls to absolute ones:
+
+```ts
+const consolidateResult(result: GlobalSearchResult, request: KibanaRequest) {
+  if(isUrlPath(result.url)) {
+    result.url = http.getAbsoluteUrl(result.url, request)
+  }
+}
+```
+
+#### Redirecting to a result
+
+Having absolute urls in GS results is a necessity for external consumers, and makes the API more consistent than mixing
+relative and absolute urls, however this makes it less trivial for UI consumers to redirect to a given result in a SPA
+friendly way (using `application.navigateTo` instead of triggering a full page refresh).
```

Comment:
> I'm not sure if we have use cases where relative and absolute URL's might be mixed

We will, in a future version of the search bar, when users will be able to search for objects on another cloud instances for example.

Reply:
What I meant was, wouldn't it be helpful to have this logic inside `navigateTo` so that other plugins can also leverage this SPA friendly navigation behaviour for other use cases.